src/blocks/posts/deprecated.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Internal dependencies
+ */
+import metadata from './block.json';
+
+const { attributes } = metadata;
+
+export default [
+	{
+		attributes: {
+			...attributes,
+			categories: {
+				type: 'string',
+			},
+		},
+		supports: {
+			align: true,
+			html: false,
+		},
+		migrate: ( oldAttributes ) => {
+			// This needs the full category object, not just the ID.
+			return {
```

Comment:
Let's query the REST API for this additional information (left a code snippet in another comment).

Reply:
Having issues getting the fetch results saved properly. There seems to be a race condition of sorts where this migration function will not await the fetch results.

Reply:
Marking this resolved since we've decided this missing data is a minor issue which does not break the actual functionality of the block.